config/initializers/timber.rb

```diff
@@ -19,6 +19,15 @@
   rack_request.path.match?(/^\/page_views\/\d{1,9}/)
 end
 
+config.integrations.rack.user_context.custom_user_hash = lambda do |rack_env|
+  session_user_id = rack_env["rack.session"].to_h["warden.user.user.key"]&.first&.first
+  if session_user_id
+    {
+      id: session_user_id
```

Comment:
can we change this key to be `user_id` so that it is clear what kind of ID it is?

Reply:
This will actually log within a user object, and everything sent here (`User::Context` middleware) is user-related, so I think `id` is still appropriate.

<img width="535" alt="Screen Shot 2019-12-02 at 1 00 05 PM" src="https://user-images.githubusercontent.com/3102842/69983165-09ff4c00-1504-11ea-9e7b-bb0870e930f1.png">

I think we should probably maintain compatibility, all else equal, right now with the current structure so that we can do this query

<img width="143" alt="Screen Shot 2019-12-02 at 1 07 02 PM" src="https://user-images.githubusercontent.com/3102842/69983433-ade8f780-1504-11ea-9cab-79a9f990087f.png">

And have it return all the appropriate data. But duly noted on this as something to keep top of mind in general.

This is how I interpret the whole thing, let me know if I'm off base.

Reply:
That is perfect! I was not aware it was already in a context thanks for the Timber learnings 😊 👍 from me!